src/Symfony/Component/Validator/Tests/Constraints/RgbColorValidatorTest.php

```diff
@@ -0,0 +1,292 @@
+
+        new RgbColor(['alphaPrecision' => 'foo']);
+    }
+
+    public function testUnexpectedType()
+    {
+        $this->expectException(UnexpectedTypeException::class);
+        $this->expectExceptionMessage('Expected argument of type "Symfony\Component\Validator\Constraints\RgbColor", "Symfony\Component\Validator\Constraints\Email" given');
+
+        $this->validator->validate(null, new Email());
+    }
+
+    public function testUnexpectedValue()
+    {
+        $this->expectException(UnexpectedValueException::class);
+        $this->expectExceptionMessage('Expected argument of type "string", "stdClass" given');
+
+        $this->validator->validate(new \stdClass(), new RgbColor());
+    }
+
+    public function testNullIsValid()
+    {
+        $this->validator->validate(null, new RgbColor());
+
+        $this->assertNoViolation();
+    }
+
+    public function testEmptyStringIsValid()
+    {
+        $this->validator->validate('', new RgbColor());
+
+        $this->assertNoViolation();
+    }
+
+    /**
+     * @dataProvider getValidRgbColorsWithDefaultOptions
+     */
+    public function testValidRgbColorsWithDefaultOptions($color)
+    {
+        $this->validator->validate($color, new RgbColor());
+
+        $this->assertNoViolation();
+    }
+
+    public function getValidRgbColorsWithDefaultOptions()
+    {
+        return [
+            ['rgb(0,0,0)'],
+            ['rgb(255,255,255)'],
+            ['RGB(0,0,0)'],
+            ['RgB(0,0,0)'],
+            ['rgb(0, 0, 0)'],
+            ['rgb(0,    0,   0)'],
+            ['rgba(0,0,0,0.2)'],
+            ['rgba(0,0,0,.75)'],
+            ['rgba(0,0,0,1)'],
+            [new class() {
+                public function __toString()
+                {
+                    return 'rgb(0,0,0)';
+                }
+            }],
+        ];
+    }
+
+    /**
+     * @dataProvider getInvalidRgbColorsWithDefaultOptions
+     */
+    public function testInvalidRgbColorsWithDefaultOptions($color)
+    {
+        $this->validator->validate($color, new RgbColor(['message' => 'foo']));
+
+        $this->buildViolation('foo')
+            ->setParameter('{{ value }}', '"'.(string) $color.'"')
+            ->setCode(RgbColor::INVALID_FORMAT_ERROR)
+            ->assertRaised();
+    }
+
+    public function getInvalidRgbColorsWithDefaultOptions()
+    {
+        return [
+            ['rgb(a,0,0)'],
+            ['rgb(0,a,0)'],
+            ['rgb(0,0,a)'],
+            ['rgb(0,0,0'],
+            ['rgb0,0,0)'],
+            ['rgb0,0,0'],
+            [' rgb(0,0,0)'],
+            ['rgb (0,0,0)'],
+            ['rgbb(0,0,0)'],
+            ['rgba(0,0,0,1.2)'],
+            ['rgba(0,0,0,0.751)'],
+            ['rgba(0,0,0,.)'],
+            ['rgba(0,0,0,3)'],
+            ['rgb(-1,0,0)'],
+            ['rgb(0,256,0)'],
+            ['rgb(0,0,256)'],
+            [new class() {
+                public function __toString()
+                {
+                    return 'rgb(0,0,0) ';
```

Comment:
How can this be part of invalid and valid?

Reply:
Valid has `'rgb(0,0,0)'` and invalid `'rgb(0,0,0) ' ` (with extra space at the end).